### ISSUE TITLE
`()` not accepted for bats-core `@test` functions

### ISSUE THREAD

**Issue by jansorg** [koalaman/shellcheck#2587]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1073, SC1036, SC1072
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bats

@test test_name() {
    echo test
}
```

#### Here's what shellcheck currently says:
```
Line 3	SC1073: Couldn't parse this bats @test. Fix to allow more checks.
Line 3	SC1036: '(' is invalid here. Did you forget to escape it?
Line 3	SC1072:  Fix any mentioned problems and try again.
```
#### Here's what I wanted or expected to see:

bats-core support definitions via `@test`  and parentheses. The `()` part seems to be unsupported by ShellCheck.

As far as I understand, bats-core's regex to find test function is here: https://github.com/bats-core/bats-core/blob/008e8ce922938c5ba4ae1ca7c3ed0a26119c3452/libexec/bats-core/bats-preprocess#L34

**Comment by koalaman** [comment]
It apparently supports arbitrary gibberish as long as it's terminated with ` {`:

```
@test )!(*%^*$ { &@!$   {
  true
}
```

ShellCheck now does the same. Thanks! Bats happens to put the contents unescaped in double quotes in the test body, so `@test #@%*@$foo%!@ {` will end up expanding `$foo` when the test is run, but as a result, ShellCheck now treats it as a literal string instead.
